subprojects/execution/src/main/java/org/gradle/internal/execution/steps/CachingContext.java

```diff
@@ -18,7 +18,7 @@
 
 import org.gradle.internal.execution.caching.CachingState;
 
-public interface CachingContext extends BeforeExecutionContext {
+public interface CachingContext extends ValidationContext {
```

Comment:
We might want to rename the contexts at some point, and this one might better be called already `ValidatedContext`... WDYT?